### ISSUE TITLE
MacOS beta missing

### ISSUE THREAD

**Issue by durka** [rust-lang/rust#52686]
Rustup reports that all of the components are discontinued when trying to update beta. Nightly updates successfully. This doesn't appear to happen on Linux either.

```
$ rustup update beta
info: syncing channel updates for 'beta-x86_64-apple-darwin'
info: latest update on 2018-07-25, rust version 1.28.0-beta.14 (0aaa819fe 2018-07-24)
warning: component 'rustc' is not available anymore on target 'x86_64-apple-darwin'
warning: component 'rust-std' is not available anymore on target 'x86_64-apple-darwin'
warning: component 'cargo' is not available anymore on target 'x86_64-apple-darwin'
warning: component 'rust-docs' is not available anymore on target 'x86_64-apple-darwin'
warning: component 'rust-src' is not available anymore
warning: component 'rustfmt-preview' is not available anymore on target 'x86_64-apple-darwin'
warning: component 'rust-std' is not available anymore on target 'wasm32-unknown-emscripten'
warning: component 'rust-std' is not available anymore on target 'asmjs-unknown-emscripten'
error: some components unavailable for download: 'rustc', 'cargo'
```

**Comment by pietroalbini** [comment]
cc @rust-lang/infra 
1.28 will be released in a week.

**Comment by pietroalbini** [comment]
Looking at [the manifest](https://static.rust-lang.org/dist/channel-rust-stable.toml), these targets have all their components marked as `available = false`:

Tier 1 platforms (tests guaranteed to pass):

* `x86_64-apple-darwin`
* `i686-apple-darwin`

Tier 2 platforms (builds guaranteed to be available):

* `aarch64-apple-ios`
* `armv7-apple-ios`
* `armv7s-apple-ios`
* `i386-apple-ios`
* `x86_64-apple-ios`

So yeah, everything Apple is broken. *Build different.*

**Comment by pietroalbini** [comment]
Also, these tier 2 platforms are not available in stable nor beta:

* `aarch64-unknown-cloudabi`
* `armv7-unknown-cloudabi-eabihf`
* `i686-unknown-cloudabi`
* `powerpc-unknown-linux-gnuspe` (not in the [platform support](https://forge.rust-lang.org/platform-support.html), but I guess tier 2)
* `sparc-unknown-linux-gnu` (not in the [platform support](https://forge.rust-lang.org/platform-support.html), but I guess tier 2)

**Comment by pietroalbini** [comment]
Builds for Apple platforms [seems to be available](https://static.rust-lang.org/dist/2018-07-20/rust-std-beta-x86_64-apple-darwin.tar.xz), so this is probably a bug in the manifest generation.

**Comment by alexcrichton** [comment]
Erm... Looking at the [Travis build](https://travis-ci.org/rust-lang/rust/builds/407712859?utm_source=github_status&utm_medium=notification) the OSX workers are... just gone!

**Comment by alexcrichton** [comment]
This... may be a travis bug? Our [Travis configuration](https://github.com/rust-lang/rust/blob/0aaa819fea00a6bf2b1303b9eb47f142f099cf5a/.travis.yml#L27-L39) for that commit indeed says `if: branch = auto` but the [build logs](https://travis-ci.org/rust-lang/rust/builds/407712859?utm_source=github_status&utm_medium=notification) confirm that it was built on the `auto` branch...

Er wait, oh dear!

...

Oh dear!

We're not building mac builds any more at all -- https://travis-ci.org/rust-lang/rust/builds/407985371

**Comment by svenfuchs** [comment]
This should now be fixed.

We have recently shipped a new parser for env vars in order to address performance issues with the previous regex based version that did not perform well under very specific conditions. 

This new parser has been tripped up by the extra newline that your env var strings have due to the YAML syntax used on the `osx` jobs: https://github.com/rust-lang/rust/blob/6a3db033ad05f156281d50ee489d727ee0e5d767/.travis.yml#L31 

Our test suite did not account for this case, and the parser blew up, causing the condition to reject the respective jobs.

We've shipped a fix for this, and this looks like it confirms we should now respond correctly to your build config, too (these builds ran on staging, but we've also confirmed this in production): 

* config: https://github.com/svenfuchs/test-2/blob/auto/.travis.yml
* before the fix: https://staging.travis-ci.org/svenfuchs/test-2/builds/725121
* after the fix: https://staging.travis-ci.org/svenfuchs/test-2/builds/725124

Again, sorry about the issue, and thank you for your patience!
